### PR TITLE
Compatibility with phpspec 3

### PR DESCRIPTION
**Pull request by pocky**
Same as https://github.com/richardmiller/BehatSpec/pull/11 but for phpspec3 :)
Integrates all changes from @richardmiller

### CODE REVIEW

File: src/Extension/ExemplifyExtension/README.md [hunk 1]
```diff
@@ -28,7 +28,7 @@ This can then be modified to describe the behaviour for the method.
 
 Requires:
 
-* PhpSpec 2.0+
+* PhpSpec 3.0+
 * PHP 5.4+
 
 To use 'named constructor' examples , you need to use latest phpspec 2.1@dev.
```
Comment (richardmiller) [inline src/Extension/ExemplifyExtension/README.md]: This could be removed now as well I think.

Comment (pocky) [inline src/Extension/ExemplifyExtension/README.md]: I think it's may be useful if an user wants to use Exemplify as standalone component

**Comment by pocky** [comment]
@richardmiller This PR should not be merged before we split and release master

**Comment by richardmiller** [comment]
Ok cool, is there anything else that needs doing before doing the split and a release?

**Pull request merged**